src/WithBackgroundImage/index.js

```diff
@@ -1,26 +1,67 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import PropTypes from 'prop-types';
 
 import useBaseUrl from '@docusaurus/useBaseUrl';
 import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import useThemeContext from '@theme/hooks/useThemeContext';
 
+import classnames from 'classnames';
+import styles from './styles.module.css';
+
+const getBackgroundImages = (imageDark, imageDarkHover, imageLight, imageLightHover) => {
+  if (!imageLight) {
+    return {
+      backgroundImage: '',
+      hoverBackgroundImage: '',
+    };
+  }
+
+  const {isDarkTheme} = useThemeContext();
+
+  const backgroundImage = isDarkTheme && imageDark ? imageDark : imageLight;
+
+  let hoverBackgroundImage = isDarkTheme && imageDarkHover ? imageDarkHover : imageLightHover;
+  hoverBackgroundImage = hoverBackgroundImage ? hoverBackgroundImage : backgroundImage;
+
+  return {
+    backgroundImage: useBaseUrl(backgroundImage),
+    hoverBackgroundImage: useBaseUrl(hoverBackgroundImage),
+  };
+};
+
 const WithBackgroundImage = ({
   children,
+  className,
   imageDark,
-  imageLight, 
-  tag: Tag, 
+  imageDarkHover,
+  imageLight,
+  imageLightHover,
+  tag: Tag,
   ...props
 }) => {
-  const {isDarkTheme} = useThemeContext();
+  const {backgroundImage, hoverBackgroundImage} =
+    getBackgroundImages(imageDark, imageDarkHover, imageLight, imageLightHover);
+  const backgroundImageStyle = {
+    '--background-image': `url('${backgroundImage}')`,
+    '--hover-background-image': `url('${hoverBackgroundImage}')`
+  };
+  const images = [imageDark, imageDarkHover, imageLight, imageLightHover]
+    .filter(url => url)
+    .map(url => useBaseUrl(url));
 
-  const backgroundImage = isDarkTheme && imageDark ? imageDark : imageLight;
-  const backgroundImageStyle = backgroundImage 
-    ? { backgroundImage: `url('${useBaseUrl(backgroundImage)}')` } 
-    : {};
+  useEffect(() => {
+    images.forEach(url => {
+      const image = new Image();
+      image.src = url;
+    });
+  }, []);
```

Comment:
This preloads the images so that there is no delay when switching the image. A nice benefit of this code is that I also included a preload for the dark mode/light mode images, meaning that switching between dark and light mode will be faster since the images have already been requested.

And to clarify, this is a preload (of sorts), so it will happen after the page has been rendered, therefore not slowing down the initial load

Reply:
Also, I've changed the code to the following 
```
window.preloadedImages = window.preloadedImages || [];
window.preloadedImages[Math.random()] = images.map(url =>
  new Image().src = url
);
```

Reply:
Actually, I may need to have these be inline svgs, because the behavior is buggy on safari. I'm going to mark this PR as a WIP in the meantime